posthog/src/main/java/com/posthog/android/PostHogFeatureFlags.java

```diff
@@ -0,0 +1,228 @@
+    }
+
+    public String getFeatureFlag(final @NonNull String key, final @Nullable Map<String, Object> options) {
+        if (!this.featureFlagsLoaded) {
+            throw new IllegalStateException(String.format("getFeatureFlag for key %s failed. Feature flags didn't load in time.", key));
+        }
+        String flagValue = this.getFlagVariants().getString(key);
+        if ((Boolean) options.get("send_event") && !this.flagCallReported.get(key)) {
+            this.flagCallReported.put(key, true);
+            this.posthog.capture(
+                    "$feature_flag_called",
+                    new Properties()
+                            .putValue("$feature_flag", key)
+                            .putValue("$feature_flag_response", flagValue));
+        }
+        return flagValue;
+    }
+
+    public Boolean isFeatureEnabled(final @NonNull String key, final @Nullable Map<String, Object> options) {
+        if (!this.featureFlagsLoaded) {
+            throw new IllegalStateException(String.format("isFeatureEnabled for key %s failed. Feature flags didn't load in time.", key));
+        }
+        return this.getFeatureFlag(key, options) != null;
+    }
+
+    /**
+     * Reloads feature flags asynchronously.
+     * <p>
+     * Constraints:
+     * <p>
+     * 1. Avoid parallel requests
+     * 2. Delay a few milliseconds after each reloadFeatureFlags call to batch subsequent changes together
+     * 3. Don't call this during initial load (as /decide will be called instead), see PostHog.java
+     */
+    public void reloadFeatureFlags() {
+        if (!this.reloadFeatureFlagsQueued) {
+            this.reloadFeatureFlagsQueued = true;
+            this.startReloadTimer();
+        }
+    }
+
+    private void startReloadTimer() {
+        if (this.reloadFeatureFlagsQueued && !this.reloadFeatureFlagsInAction) {
+            new Thread(() -> {
+                try {
+                    Thread.sleep(5);
+                    this.reloadFeatureFlagsQueued = false;
+                    this.reloadFeatureFlagsRequest();
+                } catch (Exception e) {
+                    System.err.println(e);
+                }
+            }).start();
+        }
+    }
+
+    private void setReloadingPaused(Boolean isPaused) {
+        this.reloadFeatureFlagsInAction = isPaused;
+    }
+
+    private void resetRequestQueue() {
+        this.reloadFeatureFlagsQueued = false;
+    }
+
+    private void receivedFeatureFlags(HashMap response) {
+        ValueMap flags = (ValueMap) response.get("featureFlags");
+        Persistence persistence = this.posthog.persistenceCache.get();
+
+        if (flags != null) {
+            persistence.putEnabledFeatureFlags(flags);
+            persistence.putActiveFeatureFlags(new ArrayList(flags.keySet()));
+        } else {
+            persistence.put(ACTIVE_FEATURE_FLAGS_KEY, null);
+            persistence.put(ENABLED_FEATURE_FLAGS_KEY, null);
+        }
+    }
```

Comment:
I'm a bit confused why we have "active" and "enabled" feature flags?

In this PR, `activeFeatureFlags` is unused as far as I can see and in the [iOS PR](https://github.com/PostHog/posthog-ios/pull/28/files#diff-e019e5b4b296ff0290e7a2a62105d48e408f9b5c62ea296954e3991269db6570R228) it is the opposite and `enabledFeatureFlags` is  unused.

As far as I understand the `enabled` idea only exists on the frontend where you can override the values to turn them off. If that is the funtionality we might want I would vote for removing one of these settings and maybe just stick with `activefeatureFlags` allowing us to later add `overriddenFeatueFlags` or something containing a map of any values that were explicitly overridden by the caller.

Quite possible I've misunderstood something so, so please let me know :D

Reply:
That makes sense! I would suggest that we keep `enabledFeatureFlags` over `activeFeatureFlags` since the former carries multivariate value data that is lost in `activeFeatureFlags`. Also I dont there'll be a need to also store `activeFeatureFlags` (when we end up implementing overridden flags) since it can be deduced from `enabledFeatureFlags` or via the `getFlags()` method.